Holds API

beefoo opened the pull request:
I implemented a basic hold request POST to the Holds API after patron is authenticated.  Also added an oauth redirect if patron is not logged in after pressing "place hold".

I'm running into some issues with setting local props during the login process, and was hoping to get your help w/ that as I'm not as familiar w/ React.  This is the basic test:

1. Make sure you are logged out
2. Do a search, click through to an item, click place hold
3. You should be redirected to login
4. You should be redirected back to hold request page **_(that's where I'm getting an error - for some reason if I navigate away and back to the holds request page it's fine)_**
5. Then you should be able to click "Submit your item hold request" which will do a POST to the Holds API, then redirect you to the hold confirmation page.  _Note you may get an error if I already placed a hold on a particular item; I'll resolve that in a separate issue/release_

I have not integrated Locations yet; wanted to get your thoughts on efficient implementation since it will require a few API calls

EdwinGuzman commented:
Step 4 works for me. I get redirected back to the hold request page without any issues. I do see the error in Step 5 but I've tried to request a hold on multiple items and it keeps occurring. The terminal gives me this:

```
Error calling Holds API : Method not allowed. Must be one of: OPTIONS
```

and the url also displays the error `http://local.nypl.org:3001/hold/request/pb109303-pi117281?errorMessage=Method%20not%20allowed.%20Must%20be%20one%20of:%20OPTIONS`.

beefoo commented:
Yeah I see that too... I think I'm not constructing the item id correctly... submitting a fix soon..

beefoo commented:
Okay, I think I fixed it by removing the trailing slash in the Holds API endpoint.  I also added the body-parser to read the POST data.  You'll have to `npm install` before testing

EdwinGuzman commented:
Awesome! Works well!
